core/src/main/java/fr/sncf/osrd/api/PathfindingResult.java

```diff
@@ -86,15 +135,8 @@ private static RoutePathResult makeRouteResult(
                     truncatedRange.getStart(),
                     truncatedRange.getStop()
             ));
-            if (pathfindingResult.pathWaypoints.isEmpty()) {
-                // add the first waypoint
-                var firstLocation = truncatedRange.offsetLocation(0);
-                pathfindingResult.addStep(
-                        new PathWaypointResult(firstLocation.track().getID(), firstLocation.offset())
-                );
-            }
 
-            // Add waypoints: OP as suggestions, and start / end locations
+            // Add operational points as optional waypoints
             for (var op : truncatedRange.getOperationalPoints())
                 pathfindingResult.addStep(new PathWaypointResult(op.element(), trackSection));
```

Comment:
this is unrelated to the makeRoute method, it should really be moved to a more meaningful place